src/crash/sentry.zig

```diff
@@ -101,7 +101,10 @@ fn initThread(gpa: Allocator) !void {
     sentry.c.sentry_options_set_before_send(opts, beforeSend, null);
 
     // Determine the Sentry cache directory.
-    const cache_dir = try internal_os.xdg.cache(alloc, .{ .subdir = "ghostty/sentry" });
+    const cache_dir = if (builtin.os.tag == .macos)
+        try internal_os.macos.cacheDir(alloc, "ghostty/sentry")
```

Comment:
The designated directory under `Caches` should be the bundle identifier, i.e. `com.mitchellh.ghostty`, which can be retrieved via the main bundle's [bundleIdentifier](https://developer.apple.com/documentation/foundation/nsbundle/1418023-bundleidentifier).

Otherwise we may end up with both `~/Library/Caches/com.mitchellh.ghostty` and `~/Library/Caches/ghostty` exist at some point, as some system APIs will create the former upon needed.

See [Determining Where to Store Your App-Specific Files](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/AccessingFilesandDirectories/AccessingFilesandDirectories.html):
> Use the Caches directory constant [NSCachesDirectory](https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nscachesdirectory), appending your <bundle_ID> directory for cached data files or any files that your app can re-create easily.

Reply:
https://github.com/ghostty-org/ghostty/pull/3458#discussion_r1898809993

Reply:
You should avoid writing files at non-designated locations at the first place. As a terminal focusing on native experiences, writing files at arbitrary locations is not acceptable. If dynamically looking for the bundle identifier is out of scope for this pull request, you can simply replace `ghostty` by `com.mitchellh.ghostty` in the hard-coded path for now anyway.

Reply:
Speaking of behavioral changes, moving from `~/.cache/ghostty` to `~/Library/Caches/ghostty` is already making a behavioral change. However, the former follows the XDG spec, while the latter does not follow the common practice [as noted by Apple](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/AccessingFilesandDirectories/AccessingFilesandDirectories.html). If you're not making this right now, then it likely means never. Even worse, if you decide to migrate to the bundle identifier later, and without any migration or cleanup, existing users will have leftover files at both `~/.cache/ghostty` and `~/Library/Caches/ghostty`.

Reply:
https://github.com/ghostty-org/ghostty/pull/3458#:~:text=Questions%20for%20reviewers%3A

Reply:
Since you are already making a behavioral change, and there are some unknowns regarding leftovers, why not just choose the appropriate path for now?